internal/charmstore/migrations_test.go

```diff
@@ -31,7 +29,60 @@ var _ = gc.Suite(&migrationsSuite{})
 func (s *migrationsSuite) SetUpTest(c *gc.C) {
 	s.IsolatedMgoSuite.SetUpTest(c)
 	s.db = StoreDatabase{s.Session.DB("migration-testing")}
-	s.executed = []string{}
+	s.executed = nil
+}
+
+var (
+	// migrationFields holds the fields added to mongodoc.Entity,
+	// keyed by the migration step that added them.
+	migrationEntityFields = map[mongodoc.MigrationName][]string{}
+
+	// initialFields holds all the mongodoc.Entity fields
+	// at the dawn of migration time.
+	initialEntityFields = []string{
```

Comment:
No, it can't get out of sync because by definition it represents a moment in history. It records the fields as they are now - this is the definitive information.